tests/utils.py

```diff
@@ -274,11 +274,52 @@ def clean_scsi_debug(scsi_debug_dev):
     except:
         pass
 
-def find_nvme_ctrl_devs_for_subnqn(subnqn):
+def _wait_for_nvme_controllers_ready(expected_count=1, timeout=2):
+    """
+    Wait for expected number of NVMe controllers to be in live state
+    
+    :param int expected_count: expected number of live controllers
+    :param int timeout: timeout in seconds (default: 2)
+    :returns: True if ready, False if timeout
+    """
+    start_time = time.time()
+    wait_time = 0.1  # Start with 100ms
+    max_wait = 0.5   # Cap at 500ms
+    
+    while time.time() - start_time < timeout:
+        try:
+            live_controllers = set()  # Use set for automatic deduplication
+            for ctrl_path in glob.glob("/sys/class/nvme/nvme*/"):
```

Comment:
You need to check only those controllers you're waiting for. I.e. the controller's subsystem nqn should match the expected test NQN.